File: Ravi/Microsoft/Day 5/MinimumDeletionsToMakeArrayDivisible.java
class Solution {
    
    int gcd(int a,int b) {
	    if(b == 0) return a;
	    return gcd(b, a % b);
    }
    
    public int minOperations(int[] nums, int[] numsDivide) {
        int n = numsDivide.length;
        int g = numsDivide[0];
        
        for(int i = 1; i < n; i++) g = gcd(g, numsDivide[i]);
        
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        int ans = 0;
        HashMap<Integer, Integer> hm = new HashMap<>();
        
        for(int i : nums) hm.put(i, hm.getOrDefault(i, 0) + 1);
        
        for(Integer i : hm.keySet()) {
            pq.add(i);
        }
        
        while(!pq.isEmpty()) {
            
            int gg = pq.poll();
            
            if(g % gg == 0) return ans;
            
            ans += hm.get(gg);
            
        }
        
        return -1;
        
    }
}

File: RAO.sh/Microsoft/Day V/Minimum Deletions to Make Array Divisible/MinimumDeletionsToMakeArrayDivisible.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, vector<int>& numsDivide) {
        int H = 0;
        for(int i=0; i<numsDivide.size(); i++) H = __gcd(H, numsDivide[i]);
        
        sort(nums.begin(), nums.end());
        
        int div = 0;
        for(int i=0; i<nums.size(); i++){
            if(H%nums[i]) div++;
            else return div;
        }
        
        return -1;
    }
};
